analysers/analyser_osmosis_way_angle.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+#-*- coding: utf-8 -*-
+
+###########################################################################
+##                                                                       ##
+## Copyrights Tristram Gräbener 2023                                     ##
+##                                                                       ##
+## This program is free software: you can redistribute it and/or modify  ##
+## it under the terms of the GNU General Public License as published by  ##
+## the Free Software Foundation, either version 3 of the License, or     ##
+## (at your option) any later version.                                   ##
+##                                                                       ##
+## This program is distributed in the hope that it will be useful,       ##
+## but WITHOUT ANY WARRANTY; without even the implied warranty of        ##
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         ##
+## GNU General Public License for more details.                          ##
+##                                                                       ##
+## You should have received a copy of the GNU General Public License     ##
+## along with this program.  If not, see <http://www.gnu.org/licenses/>. ##
+##                                                                       ##
+###########################################################################
+
+# This analyser is similar to Analyser_Osmosis_Way_Approximate
+# However, this one measures angles and not radiuses
+# It most likely indicates a node that was accidentaly dragged
+# While Analyser_Osmosis_Way_Approximate indicates that the way was drawn
+# with too few nodes
+
+from modules.OsmoseTranslation import T_
+from .Analyser_Osmosis import Analyser_Osmosis
+
+sql10 = """
+WITH
+    points AS (
+        SELECT
+            id,
+            tags->'{tag}' AS type,
+            (ST_DumpPoints(linestring)).geom AS geom4326,
+            (ST_DumpPoints(ST_Transform(linestring, {srid}))).geom AS geom,
+            (ST_DumpPoints(ST_Transform(linestring, {srid}))).path[1] AS index
+        FROM
+            {table_prefix}ways as ways
+        WHERE
+            tags != ''::hstore AND
+            tags?'{tag}' AND tags->'{tag}' IN ('{tag_values}') AND
+            ST_NPoints(linestring) >= 3
+    ),
+    angles AS (
+        SELECT
+            id, index, type,
+            lead(geom4326) OVER (PARTITION BY id ORDER BY index) AS geom4326,
+            lead(geom) OVER (PARTITION BY id ORDER BY index) AS geom,
```

Comment:
Could be removed.

```suggestion
            lead(geom) OVER (PARTITION BY id ORDER BY index) AS geom,
```

Reply:
I’m don’t understand the suggested change, it looks the same. What could be removed?

Reply:
The fix suggestion is strange indeed, but I think @frodrigo means that `geom` isn't used, i.e. line 52 can be removed

Reply:
Thank you! It’s removed